assayist/client/query.py

```diff
@@ -15,6 +15,66 @@ def set_connection(neo4j_url):
     neomodel.db.set_connection(neo4j_url)
 
 
+def get_container_by_component(component_name, component_version):
+    """Query for builds by component name and version.
+
+    Finds builds that produced container images which include or embed a component with the
+    specified name and version, and any version preceding that version.
+
+    :param str component_name: name of the component to query for
+    :param str component_version: version of the component to query for
+    :return: list of builds IDs
+    :rtype: list
+    """
+    query = """
+        /// Find a component and a sourcelocation with the specified name and version. These can
+        /// be emdedded in other sourcelocation that are used as a source for a build or directly
+        /// used for a build.
+        MATCH (c:Component {{canonical_name: '{name}'}}) <-[:SOURCE_FOR]-
+              (version_sl:SourceLocation {{canonical_version: '{version}'}})
+              <-[:UPSTREAM|EMBEDS*0..]- (target_sl:SourceLocation)
+              <-[:BUILT_FROM]- (direct_builds:Build)
```

Comment:
If you put a `(:SourceLocation) <-[:SUPERSEDES*0..]-` before the `target_sl` node, won't you get all the builds you want without needing to use `WITH`?

Reply:
Right, that would work if the SUPERSEDES relationship was defined between the dist-git SLs only since only those will then have a BUILD_FROM relationship with a build. I assumed here that SUPERSEDES would be between upstream SLs, which requires that extra hop from upstream SL to dist-git SL to build (if that makes sense).

If I rework the query to use SUPERSEDES on dist-git SLs, then yes, the WITH can be removed.